src/driver/odoc_driver.ml

```diff
@@ -551,8 +551,10 @@ let remap_virtual_interfaces duplicate_hashes pkgs =
       })
     pkgs
 
+type action_mode = CompileOnly | LinkAndGen | All
```

Comment:
I would find something like `type action_mode = Compile | Link | Gen` more future-proof, and we don't have too many actions so it's not too tedious to enumerate them all in a pattern-matching anyway.
The cli arg would be something like:
- `--actions compile`
- `--actions link,gen`
- `--actions compile,link,gen` (maybe use an alias `--actions all` for the full list)

Would that be problematic if users specify an invalid combination like `--actions compile,gen` ? That could be filtered when parsing the arg anyway.

Reply:
I don't anticipate users using this feature at all - it's really only here so the docs CI can drive this. As such, I don't really worry so much about future-proofing this, as whatever we do will be very tightly coupled with the docs CI anyway.